src/Altinn.AccessManagement.Persistence/DelegationMetadataRepo.cs

```diff
@@ -335,6 +335,56 @@ public async Task<DelegationChange> InsertDelegation(ResourceAttributeMatchType
             return await InsertResourceRegistryDelegation(delegationChange, cancellationToken);
         }
 
+        /// <summary>
+        ///  Fetches all instance delegated to given param
+        /// </summary>
+        /// <param name="toUuid">list of parties that has received an instance delegation</param>
+        /// <param name="cancellationToken">cancellation token</param>
+        /// <returns></returns>
+        public async Task<List<InstanceDelegationChange>> GetAllCurrentReceivedInstanceDelegations(List<Guid> toUuid, CancellationToken cancellationToken = default)
+        {
+            using var activity = TelemetryConfig.ActivitySource.StartActivity(ActivityKind.Client);
+
+            var query = /*strpsql*/ @"
+                SELECT
+                    instancedelegationchangeid
+                    ,delegationchangetype
+                    ,instanceDelegationMode
+                    ,resourceid
+                    ,instanceid
+                    ,fromuuid
+                    ,fromtype
+                    ,touuid
+                    ,totype
+                    ,performedby
+                    ,performedbytype
+                    ,blobstoragepolicypath
+                    ,blobstorageversionid
+                    ,created
+                FROM 
+                    delegation.instancedelegationchanges
+                    AND delegationchangetype != 'revoke_last'
+                WHERE
+                    touuid = ANY(@toUuid)
+                GROUP BY
+                    resourceid, touuid, fromuuid;
```

Comment:
Samme som JK dersom du ikke har instanceId i group by så vil den slå sammen alle instansene desuten er det vel ikke mulig å bruke en group by og hente ut alle kolonner for de kolenne som ikke er grupert på vil kunne være mange verdier så de må ha en agregate funksjon for å finne hvilken som skal brukes. Jeg ville brukt Common table expression her og grupere på de dataene som var viktige og så hente ut max(instancedelegationchangeid) og deretter joine med denne common table expressionen med tabellen på nytt med id som match så ville jeg få ut alle siste endringer.

Reply:
stemmer, templating før jeg fikk testet spørringen. Endret på spørringen nu